include/PriorityQueue.h

```diff
@@ -0,0 +1,35 @@
+#include <list>
+#include <stdexcept>
+#include <functional>
+
+template <typename T>
+struct MyComparator {
+    bool operator()(const T& a, const T& b) const {
+        return a < b;
+    }
+};
+
+template <typename T>
+class PriorityQueue {
+public:
+    void push(const T& t) {
+        auto it = m_queue.begin();
```

Comment:
When possible it's recommended to call `cbegin()` and `cend()`. You can never be too safe